cli/core/pkg/command/root.go

```diff
@@ -129,92 +137,20 @@ var tmcCmd = &cobra.Command{
+func addPluginsToTarget(mapCtxTypeToCmd map[cliv1alpha1.Target]*cobra.Command) error {
+	installedPlugins, standalonePlugins, err := pluginmanager.InstalledPlugins()
 	if err != nil {
 		return fmt.Errorf("unable to find installed plugins: %w", err)
 	}
 
-	if ctxType == configapi.CtxTypeK8s {
-		// Standalone plugins exist only for K8s context type.
-		for i := range standalonePlugins {
-			if standalonePlugins[i].Group == cliapi.SystemCmdGroup {
-				// Do not include plugins from the system command group.
-				continue
-			}
-			cmd.AddCommand(cli.GetCmd(&standalonePlugins[i]))
-		}
-	}
-
-	for i := range ctxPlugins {
-		cmd.AddCommand(cli.GetCmd(&ctxPlugins[i]))
-	}
-	return nil
-}
-
-func getAvailablePlugins() ([]*cliapi.PluginDescriptor, error) {
-	plugins := make([]*cliapi.PluginDescriptor, 0)
-	var err error
-
-	if config.IsFeatureActivated(cliconfig.FeatureContextAwareCLIForPlugins) {
-		currentServerName := ""
-
-		server, err := config.GetCurrentServer()
-		if err == nil && server != nil {
-			currentServerName = server.Name
-		}
-
-		serverPlugin, standalonePlugins, err := pluginmanager.InstalledPlugins(currentServerName)
-		if err != nil {
-			return nil, fmt.Errorf("find installed plugins: %w", err)
-		}
+	installedPlugins = append(installedPlugins, standalonePlugins...)
 
-		allPlugins := serverPlugin
-		allPlugins = append(allPlugins, standalonePlugins...)
-		for i := range allPlugins {
-			plugins = append(plugins, &allPlugins[i])
-		}
-	} else {
-		// TODO: cli.ListPlugins is deprecated: Use pluginmanager.AvailablePluginsFromLocalSource or pluginmanager.AvailablePlugins instead
-		plugins, err = cli.ListPlugins()
-		if err != nil {
-			return nil, fmt.Errorf("find available plugins: %w", err)
+	for i := range installedPlugins {
+		if cmd, exists := mapCtxTypeToCmd[installedPlugins[i].Target]; exists {
+			cmd.AddCommand(cli.GetCmd(&installedPlugins[i]))
 		}
 	}
-	return plugins, nil
-}
-
-func checkAndInstallMissingPlugins(plugins []*cliapi.PluginDescriptor) ([]*cliapi.PluginDescriptor, error) {
-	// check that all plugins in the core distro are installed or do so.
-	if !noInit && !cli.IsDistributionSatisfied(plugins) {
-		s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
-		if err := s.Color("bgBlack", "bold", "fgWhite"); err != nil {
-			return nil, err
-		}
-		s.Suffix = fmt.Sprintf(" %s", "initializing")
-		s.Start()
-		cfg, err := config.GetClientConfig()
-		if err != nil {
-			log.Fatal(err)
-		}
-		repos := cli.NewMultiRepo(cli.LoadRepositories(cfg)...)
-		err = cli.EnsureDistro(repos)
-		if err != nil {
-			return nil, err
-		}
-		// TODO: cli.ListPlugins is deprecated: Use pluginmanager.AvailablePluginsFromLocalSource or pluginmanager.AvailablePlugins instead
-		plugins, err = cli.ListPlugins()
-		if err != nil {
-			return nil, fmt.Errorf("find available plugins: %w", err)
-		}
-		s.Stop()
-	}
-	return plugins, nil
+	return nil
 }
 
 func duplicateAliasWarning() {
```

Comment:
This function checks for duplicate aliases between commands/plugins that are at the root level.
Now that we install plugins under the TMC target, we may want to also check for duplicate aliases at that level.
To be more future-proof, maybe it would even be better to check for duplicate aliases for each target, even though it is redundant for the K8s target (since those plugins are also at the root level, where we already check).

Reply:
This can be done in a follow-up PR and does not need to delay the merge of this PR